docs/user_manual/source/exceptions_interrupts.rst

```diff
@@ -98,6 +97,27 @@ In Debug Mode, all interrupts are ignored independent of ``mstatus``.MIE and the
    Load bus fault and store bus fault are handled as imprecise non-maskable interrupts
    (as opposed to precise exceptions).
 
+Non Maskable Interrupts
+-----------------------
+
+NMI's update ``mepc``, ``mcause`` and ``mstatus`` similar to regular interrupts. However, the contents of ``mepc`` is not guaranteed to point to the instruction after the faulted Load or Store.
+
+An NMI will occur when a Load or Store instruction experiences a bus fault. The NMI is imprecise, meaning that it will occur **after** the instruction that caused it retires.
```

Comment:
Load or Store -> load or store

Reply:
The NMI is imprecise, meaning that it will occur **after** the instruction that caused it retires.
->
The fault resulting in an NMI is handled in an imprecise manner, meaning that the instruction that causes the fault is allowed to retire and the associated NMI is taken **afterwards**.